### PR TITLE
[stdlib]Array iterator

### PR DESCRIPTION
**Pull request by lancep**
Testing a more efficient iterator for Array.

### CODE REVIEW

**Comment by lancep** [comment]
@swift-ci please benchmark

File: stdlib/public/core/Arrays.swift.gyb [hunk 1]
```diff
@@ -453,6 +453,29 @@ if True:
     raise ValueError('Unhandled case: ' + Self)
 }%
 
+@_fixed_layout
```
Comment (slavapestov) [inline stdlib/public/core/Arrays.swift.gyb]: Could this be a nested type of ${Self}, like @airspeedswift's other PR?

Comment (slavapestov) [reply stdlib/public/core/Arrays.swift.gyb]: Is this effectively a hand-coded specialization of IndexingIterator?

Comment (lancep) [reply stdlib/public/core/Arrays.swift.gyb]: Good call, will do

Comment (lancep) [reply stdlib/public/core/Arrays.swift.gyb]: Yes

**Comment by lancep** [comment]
@swift-ci please test

**Comment by lancep** [comment]
Please test with the following pull request
https://github.com/apple/swift-package-manager/pull/1505

@swift-ci please test

**Comment by lancep** [comment]
Please test with the following pull request
apple/swift-package-manager#1505

@swift-ci please test

**Pull request closed without merging**